test/unit/lib/utils/fs/safeMoveFile.test.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+
+const fse = require('fs-extra');
+const sinon = require('sinon');
+const chai = require('chai');
+const { getTmpDirPath } = require('../../../../utils/fs');
+const { join } = require('path');
+const { expect } = require('chai');
+const fsp = require('fs').promises;
+const sinonChai = require('sinon-chai');
+const safeMoveFile = require('../../../../../lib/utils/fs/safeMoveFile');
+
+chai.use(sinonChai);
+
+/**
+ * Returns and Error object that resembles the EXDEV errors that are thrown
+ * when you try to rename across device boundaries.
+ *
+ * @returns an Error object that resembles the EXDEV errors are thrown
+ */
+const createFakeExDevError = () => {
+  // Properties captured by using fs.renameSync in the Node v12.20.1 REPL
+  const fakeCrossDeviceError = new Error(
+    "Error: EXDEV: cross-device link not permitted, rename '/foo/bar' -> '/bar/baz'"
+  );
+  // This is the important property that safeMoveFile looks for
+  // to fallback to copy then rename behaviour
+  fakeCrossDeviceError.code = 'EXDEV';
+
+  // These other properties aren't used but it doesn't hurt to have an accurate error object
+  fakeCrossDeviceError.errno = -18;
+  fakeCrossDeviceError.syscall = 'rename';
+  fakeCrossDeviceError.path = '/foo/bar';
+  fakeCrossDeviceError.dest = '/bar/baz';
+  return fakeCrossDeviceError;
+};
+
+/**
+ * The name of the file that will be moved
+ */
+const artefactName = 'foo-bar.zip';
```

Comment:
nit: typo, it rather should be `artifactName`

Reply:
Oh sorry, that's Australian spelling. Fixed!